mep/people/models.py

```diff
@@ -123,19 +119,28 @@ def merge_with(self, person):
                .filter(account_people__gt=1).exists():
             raise MultipleObjectsReturned("Can't merge a person record with a shared account.")
 
-        # identify the account other events will be reassociated with
-        primary_account = person.account_set.first()
         # make sure specified person is skipped even if in the current queryset
         merge_people = self.exclude(id=person.id)
 
         for merge_person in merge_people:
-            for account in merge_person.account_set.all():
-                # reassociate all events with the main account
-                account.event_set.update(account=primary_account)
-                # reassociate any addresses with the main account
-                account.address_set.update(account=primary_account)
-                # delete the empty account
-                account.delete()
+            if merge_person.has_account():
+            # if the merged person had an account
+                if person.has_account():
+                    # *and* if the main person had an account
+                    for account in merge_person.account_set.all():
+                        # reassociate all events with the main account
+                        account.event_set.update(account=primary_account)
+                        # reassociate any addresses with the main account
+                        account.address_set.update(account=primary_account)
+                        # delete the empty account
+                        account.delete()
+                else:
+                # a merged person had an account, but the main person doesn't
+                    for account in merge_person.account_set.all():
+                        # swap the account's owner to the main person
+                        account.persons.add(person)
+                        account.persons.remove(merge_person)
```

Comment:
I think this won't work right for merging multiple people with accounts if the primary person has no account.  Won't this just aggregate all accounts on one person rather than consolidating the events on the accounts into one account on a single person record?

Would it work to move this check first and then use the existing merge logic? Check for no account on the main person record, swap with the first account (if there is one), and then consolidate with all remaining accounts & events.

Might be helpful to add a test case where you are merging multiple people and primary has no account.

Reply:
Ah, this makes sense - I had been thinking about it in terms of "merge the people and move everything else" but really merging people should also merge the accounts, which necessitates moving all the accounts' events. I'll just use the existing logic.